src/Apps/CollectorProfile/Routes/Saves2/Components/CreateNewListModal/CreateNewListModalWizard.tsx

```diff
@@ -37,3 +50,38 @@ export const CreateNewListModalWizard: FC<CreateNewListModalWizardProps> = ({
     />
   )
 }
+
+export interface CreateNewListModalWizardQueryRendererProps {
+  onComplete: (artworkList: ArtworkList) => void
+  onClose: () => void
+}
+
+export const CreateNewListModalWizardQueryRenderer: FC<CreateNewListModalWizardQueryRendererProps> = props => {
+  return (
+    <SystemQueryRenderer<CreateNewListModalWizardQuery>
+      query={graphql`
+        query CreateNewListModalWizardQuery {
+          me {
+            collection(id: "saved-artwork") {
+              artworksConnection {
+                totalCount
+              }
+            }
+          }
+        }
+      `}
+      render={({ props: relayProps, error }) => {
+        if (error) {
+          console.error(error)
+          return null
+        }
+
+        if (!relayProps?.me) {
```

Comment:
In this code there is a small problem with the fact that the modal can be displayed with In this code there is a small problem with the fact that the modal can be displayed with a delay when the user has a bad Internet connection when the user has a bad Internet connection (see attached video). There are 2 ways to solve this problem:
1. Query and pass the artworks count for the `ArtworkListsHeader` in [the parent component](https://github.com/artsy/force/blob/28e7b7ae9887a4e5789c3c38e8dd88dae3356088/src/Apps/CollectorProfile/Routes/Saves2/CollectorProfileSaves2Route.tsx#L59)
2. Don't wait until a response is received from the backend and display the modal

IMHO the first way will be better 🤔


https://user-images.githubusercontent.com/3513494/230112626-0a8293cd-3379-44c4-88e5-3df26f9aa48c.mp4

Reply:
thank you, good advice!